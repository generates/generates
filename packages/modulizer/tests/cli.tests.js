const { readFileSync } = require('fs')
const { test } = require('@ianwalter/bff')
const execa = require('execa')

test('cjs file not generated when --esm specified', async ({ expect }) => {
  const input = 'tests/fixtures/exportDefaultFunction.js'
  const esm = 'tmp/one.js'
  const args = [input, '--esm', esm]
  const { stdout } = await execa('./cli.js', args)
  expect(stdout).toMatchSnapshot()
  expect(readFileSync(esm, 'utf8')).toMatchSnapshot()
})

test('dist file is transpiled when --babel passed', async ({ expect }) => {
  const input = 'tests/fixtures/exportDefaultNewExpression.js'
  const cjs = 'tmp/two.js'
  const args = [input, '--cjs', cjs, '--babel']
  const { stdout } = await execa('./cli.js', args)
  expect(stdout).toMatchSnapshot()
  expect(readFileSync(cjs, 'utf8')).toMatchSnapshot()
})

test('dist file generated using custom plugins', async ({ expect }) => {
  const input = 'tests/fixtures/exportObjectWithVueComponent.js'
  const cjs = 'tmp/three.js'
  const plugins = ['--plugins', 'tests/helpers/vuePlugin.js']
  const args = [input, '--cjs', cjs, ...plugins]
  const { stdout } = await execa('./cli.js', args)
  expect(stdout).toMatchSnapshot()
  expect(readFileSync(cjs, 'utf8')).toMatchSnapshot()
})
