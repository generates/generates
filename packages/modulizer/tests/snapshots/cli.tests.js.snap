// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`cjs file not generated when --esm specified 1`] = `"ðŸ“¦  [1mWriting ES Module dist file:[22m [90mtmp/one.js[39m"`;

exports[`cjs file not generated when --esm specified 2`] = `
"function greeting () {
  return 'Hello World'
}

export default greeting;
"
`;

exports[`dist file generated using custom plugins 1`] = `"ðŸ’¿  [1mWriting CommonJS dist file:[22m [90mtmp/three.js[39m"`;

exports[`dist file generated using custom plugins 2`] = `
"'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

//
//
//
//
//
//
//

var script = {
  data: () => ({ name: 'World' })
};

function normalizeComponent(template, style, script, scopeId, isFunctionalTemplate, moduleIdentifier /* server only */, shadowMode, createInjector, createInjectorSSR, createInjectorShadow) {
    if (typeof shadowMode !== 'boolean') {
        createInjectorSSR = createInjector;
        createInjector = shadowMode;
        shadowMode = false;
    }
    // Vue.extend constructor export interop.
    const options = typeof script === 'function' ? script.options : script;
    // render functions
    if (template && template.render) {
        options.render = template.render;
        options.staticRenderFns = template.staticRenderFns;
        options._compiled = true;
        // functional template
        if (isFunctionalTemplate) {
            options.functional = true;
        }
    }
    // scopedId
    if (scopeId) {
        options._scopeId = scopeId;
    }
    let hook;
    if (moduleIdentifier) {
        // server build
        hook = function (context) {
            // 2.3 injection
            context =
                context || // cached call
                    (this.$vnode && this.$vnode.ssrContext) || // stateful
                    (this.parent && this.parent.$vnode && this.parent.$vnode.ssrContext); // functional
            // 2.2 with runInNewContext: true
            if (!context && typeof __VUE_SSR_CONTEXT__ !== 'undefined') {
                context = __VUE_SSR_CONTEXT__;
            }
            // inject component styles
            if (style) {
                style.call(this, createInjectorSSR(context));
            }
            // register component module identifier for async chunk inference
            if (context && context._registeredComponents) {
                context._registeredComponents.add(moduleIdentifier);
            }
        };
        // used by ssr in case component is cached and beforeCreate
        // never gets called
        options._ssrRegister = hook;
    }
    else if (style) {
        hook = shadowMode
            ? function (context) {
                style.call(this, createInjectorShadow(context, this.$root.$options.shadowRoot));
            }
            : function (context) {
                style.call(this, createInjector(context));
            };
    }
    if (hook) {
        if (options.functional) {
            // register for functional component in vue file
            const originalRender = options.render;
            options.render = function renderWithStyleInjection(h, context) {
                hook.call(context);
                return originalRender(h, context);
            };
        }
        else {
            // inject component registration as beforeCreate hook
            const existing = options.beforeCreate;
            options.beforeCreate = existing ? [].concat(existing, hook) : [hook];
        }
    }
    return script;
}

/* script */
const __vue_script__ = script;

/* template */
var __vue_render__ = function () {var _vm=this;var _h=_vm.$createElement;var _c=_vm._self._c||_h;return _c('div',[_c('input',{directives:[{name:\\"model\\",rawName:\\"v-model\\",value:(_vm.name),expression:\\"name\\"}],domProps:{\\"value\\":(_vm.name)},on:{\\"input\\":function($event){if($event.target.composing){ return; }_vm.name=$event.target.value;}}}),_vm._v(\\" \\"),_c('div',[_vm._v(\\"Hello \\"+_vm._s(_vm.name)+\\"!\\")])])};
var __vue_staticRenderFns__ = [];

  /* style */
  const __vue_inject_styles__ = undefined;
  /* scoped */
  const __vue_scope_id__ = undefined;
  /* module identifier */
  const __vue_module_identifier__ = undefined;
  /* functional template */
  const __vue_is_functional_template__ = false;
  /* style inject */
  
  /* style inject SSR */
  
  /* style inject shadow dom */
  

  
  const __vue_component__ = normalizeComponent(
    { render: __vue_render__, staticRenderFns: __vue_staticRenderFns__ },
    __vue_inject_styles__,
    __vue_script__,
    __vue_scope_id__,
    __vue_is_functional_template__,
    __vue_module_identifier__,
    false,
    undefined,
    undefined,
    undefined
  );

exports.Greeting = __vue_component__;
"
`;

exports[`dist file is transpiled when --babel passed 1`] = `"ðŸ’¿  [1mWriting CommonJS dist file:[22m [90mtmp/two.js[39m"`;

exports[`dist file is transpiled when --babel passed 2`] = `
"'use strict';

class SomeClass {}

var exportDefaultNewExpression = new SomeClass();

module.exports = exportDefaultNewExpression;
"
`;
